File: leetcode/153-Find-Minimum-in-Rotated-Sorted-Array/phyllis.cpp
class Solution {
public:
 
    int findMin(vector<int>& nums) {     
        int size = nums.size();        
        
        if(size == 1) 
            return nums[0];
        
        int prev = nums[0];
        for(int i = 1 ; i < size; i++){
            if(nums[i] < prev)
                return nums[i];
                
            prev = nums[i];
        }
        
        return nums[0];
    }
};

File: leetcode/238-Product-of-Array-Except-Self/phyllis.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int length = nums.length;
        int[] output = new int[length];
        if(length == 0)
            return output;      
        
        int accumulation = 1;
        for(int i = 0; i < length; i++){
            output[i] = accumulation;
            accumulation *= nums[i];
        }
        
        accumulation = 1;
        for(int i = length - 1; i >= 0; i--){
             output[i] *= accumulation;
             accumulation *= nums[i];    
        }
        
        return output;
    }
}
